Sources/SwiftSyntaxMacroExpansion/MacroExpansion.swift

```diff
@@ -201,3 +211,35 @@ fileprivate extension SyntaxProtocol {
     return formatted.trimmedDescription(matching: { $0.isWhitespace })
   }
 }
+
+func applyingAttrsModifiers(
+  to node: DeclSyntax,
+  attributes: AttributeListSyntax?,
+  modifiers: ModifierListSyntax?
+) -> DeclSyntax {
+  func _combine<C: SyntaxCollection>(_ left: C, _ right: C?) -> C? {
+    guard let right = right else { return left }
+    var elems: [C.Element] = []
+    elems.append(contentsOf: left)
+    elems.append(contentsOf: right)
+    return C(elems)
+  }
+  var node = node
+  if let attributes = attributes,
+    let withAttrs = node.asProtocol(WithAttributesSyntax.self)
+  {
+    node = withAttrs.with(
+      \.attributes,
+      _combine(attributes, withAttrs.attributes)
+    ).as(DeclSyntax.self)!
```

Comment:
You can use `cast(DeclSyntax.self)` to remove the need for the `!`.

Same below.